mrmustard/physics/ansatze.py

```diff
@@ -380,3 +380,113 @@ def __and__(self, other: PolyExpAnsatz) -> PolyExpAnsatz:
         bs = [math.concat([b1, b2], axis=-1) for b1 in self.b for b2 in other.b]
         cs = [math.outer(c1, c2) for c1 in self.c for c2 in other.c]
         return self.__class__(As, bs, cs)
+
+
+class ArrayAnsatz(Ansatz):
+    r"""Represents the ansatz as a multidimensional array."""
+
+    def __init__(self, array: Batch[Tensor]):
+        r"""Note that the array is batched."""
+        self.array = array
+
+    def __neg__(self) -> ArrayAnsatz:
+        r"""
+        Negates the values in the array.
```

Comment:
We should have another PR to fix the docs and make sure they appear in the online webpage. When we do, we should remember to remove all the docstrings for the methods in ArrayAnsatz and PolyExpAnsatz that are abstract methods in Ansatz, because sphinx is smart enough to copy/paste those coming from Ansatz. This way, we can ensure that every abstract method and their representation have one docstring, and not several (potentially conflicting) docstrings